src/adapters/fiat-exchange-rates.ts

```diff
@@ -0,0 +1,53 @@
+import { fiatExchangeRatesErrorParser, fiatExchangeRatesParser } from "src/adapters/parsers";
+import { FIAT_EXCHANGE_RATES_API_URL } from "src/constants";
+import { Currency, FiatExchangeRates } from "src/domain";
+
+export interface GetFiatExchangeRatesResponse {
+  rates: Record<string, number>;
+}
+
+export interface GetFiatExchangeRatesError {
+  error: {
+    code: string;
+    message: string;
+  };
+}
+
+interface GetFiatExchangeRatesParams {
+  apiKey: string;
+}
+
+const getFiatExchangeRates = ({
+  apiKey,
+}: GetFiatExchangeRatesParams): Promise<FiatExchangeRates> => {
+  const requestQueryParams = new URLSearchParams({
+    access_key: apiKey,
+    base: Currency.USD,
+    symbols: Object.values(Currency).join(","),
+  });
+  const requestUrl = `${FIAT_EXCHANGE_RATES_API_URL}?${requestQueryParams.toString()}`;
+
+  return fetch(requestUrl).then(async (res) => {
+    if (res.ok) {
+      const data: unknown = await res.json();
+      const parsedResponse = fiatExchangeRatesParser.safeParse(data);
+
+      if (parsedResponse.success) {
+        return parsedResponse.data.rates;
+      } else {
+        throw new Error("An error occurred when trying to parse the getFiatExchangeRates response");
```

Comment:
We should not loos track of the `parsedResponse.error`. Instead of returning this text, we should return the error, to be later decoded by any interested consumer of this API.

Reply:
Right! Fixed